sapi/apache2handler/sapi_apache2.c

```diff
@@ -484,26 +508,13 @@ php_apache_server_startup(apr_pool_t *pconf, apr_pool_t *plog, apr_pool_t *ptemp
 	if (apache2_php_ini_path_override) {
 		apache2_sapi_module.php_ini_path_override = apache2_php_ini_path_override;
 	}
-#ifdef ZTS
-	int expected_threads;
-	if (ap_mpm_query(AP_MPMQ_MAX_THREADS, &expected_threads) != APR_SUCCESS) {
-		expected_threads = 1;
-	}
-
-	php_tsrm_startup_ex(expected_threads);
-# ifdef PHP_WIN32
-	ZEND_TSRMLS_CACHE_UPDATE();
-# endif
-#endif
-
-	zend_signal_startup();
-
-	sapi_startup(&apache2_sapi_module);
-	if (apache2_sapi_module.startup(&apache2_sapi_module) != SUCCESS) {
+#ifndef PHP_WIN32
+	if (php_apache_startup_php() != OK) {
 		return DONE;
 	}
 	apr_pool_cleanup_register(pconf, NULL, php_apache_server_shutdown, apr_pool_cleanup_null);
 	php_apache_add_version(pconf);
```

Comment:
Unfortunately this accesses `PG(expose_php)`, which is controlled by ini settings. It's too early to call this as ini settings haven't been processed yet. It may be safe to move into `php_apache_startup_php()` since it's called once per process.

Reply:
As is, that is already moved into `php_apache_child_init()` for `PHP_WIN32`, and apparently works (tested with default, and both alternatives in php.ini).